src/requests/publishAssetAsync.luau

```diff
@@ -25,28 +30,83 @@ local function publishAssetAsync(
 	local existingAssetId = if existingAsset then existingAsset.assetId else nil
 	local assetDetails = if existingAssetId then getAssetDetailsAsync(existingAssetId, apiKey) else nil
 
-	local variables = runLuauTask(publishAssetTask, {
-		CREATOR_ID = environment.creatorId,
-		CREATOR_TYPE = environment.creatorType,
-		ASSET_TYPE = asset.type or "Package",
+	local creatorIdKey = if environment.creatorType == "User" then "userId" else "groupId"
```

Comment:
nit: enum / map (personal preference). though i assume creatorType is just a string and we can't be cute and type it into an enum value easily